### PR TITLE
Add some new checks and a power-monitor design

### PR DESCRIPTION
**Pull request by d-haldane**
Also removing net names in terminations - the autogenerated names can have more context than internally coded ones.

Added some simple calculations to try to get operating points of resistors and capacitors

Enabled component checks for resistors and capacitors

Updated Operating temperature to use a struct.

### CODE REVIEW

**Review by electricfungi — COMMENTED** [review]
the operating-point checks fail in power-monitor.stanza

File: designs/power-monitor.stanza [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#use-added-syntax(jitx)
```
Comment (d-haldane) [inline designs/power-monitor.stanza:1]: Add a header block saying what is is what it does, what it connects to. Note that some checks are expected to fail.

File: components/maxim/MAX1606x.stanza [hunk 2]
```diff
@@ -12,37 +12,87 @@ defpackage ocdb/maxim/MAX1606x:
...
 
-public pcb-module module :
-  inst i : component
+  property(self.rated-temperature) = RatedTemperature(Interval(-40.0, 125.0, false))
+
+  property(self.VCC.power-pin) = PowerPin(Interval(1.0, 5.5, false))
+  eval-when has-property?(self.VCC.rail-voltage):
+    val vdd = property(self.VCC.rail-voltage)
+    for p in [self.TOL self.SRT self.nMR self.WDI self.nMARGIN] do :
+      property(p.generic-pin) = GenericPin(Interval(-0.3, vdd + 0.3, false), 100.0)
```
Comment (bhusang) [inline components/maxim/MAX1606x.stanza:57]: Cool way to add voltage thresholds to pins post-facto.

**Review by bhusang — APPROVED** [review]
Looks good -- with some small changes planned for the next PR to clarify tol%()

File: components/analog-devices/ADM7150.stanza [hunk 3]
```diff
@@ -78,7 +81,7 @@ public pcb-module module (v-out:Double) :
   cap-strap(ps.vout, gnd, 10.0e-6)
   cap-strap(ps.vreg, gnd, 10.0e-6)
   cap-strap(ps.byp,  gnd, 1.0e-6)
-  cap-strap(ps.byp,  gnd, 100.0e-6)
+  cap-strap(ps.byp,  gnd, 47.0e-6)
```
Comment (ariel-hi) [inline components/analog-devices/ADM7150.stanza:84]: I'm curious -- does this indicate that a part's requirements have changed, or that the numbers were incorrect?

File: components/microsemi/A2F200M3F-FGG256I.stanza [hunk 4]
```diff
@@ -204,14 +49,14 @@ public pcb-component component :
...
     [SDD1      | T[12]  | Right | analog] 
-    [ADC[0]    | T[5]   | Right | analog] 
-    [ADC[1]    | R[6]   | Right | analog] 
-    [ADC[2]    | P[7]   | Right | analog] 
-    [ADC[3]    | M[5]   | Right | analog] 
-    [ADC[4]    | M[9]   | Right | analog] 
-    [ADC[5]    | N[9]   | Right | analog] 
-    [ADC[6]    | P[9]   | Right | analog] 
-    [ADC[7]    | R[9]   | Right | analog] 
+    [ADC[0]    | T[5]   | Left | analog]
```
Comment (ariel-hi) [inline components/microsemi/A2F200M3F-FGG256I.stanza:52]: Likewise, when pin directions change, does this indicate a change in the part requirements or a preferential change?

File: components/analog-devices/ADM7150.stanza (hunk 3 above)
Comment (d-haldane) [reply components/analog-devices/ADM7150.stanza:84]: The part has a range of acceptable values - 100uF was pretty extreme, 47uF is more reasonable.

File: components/microsemi/A2F200M3F-FGG256I.stanza (hunk 4 above)
Comment (d-haldane) [reply components/microsemi/A2F200M3F-FGG256I.stanza:52]: Preferential to get the symbols looking better.

**Pull request merged**